### PR TITLE
fix: migrator-integration-test tests not running #191

### PR DESCRIPTION
**Pull request by Roiocam**
## Motivation

Resolves: #191

## Resolves in step by step

- [x] investigation Migrator integration test compile issue.
- [x] investigation package name is issue or not.
- [x] investigation `integration/testOnly` will be an issue or not.

#### compile issue 

`migratorIntegration` should be depend on `migrator` ✅

#### package name issue & sbt task issue

- `sbt ++2.13 "clean; integration/testOnly org.apache.pekko.persistence.jdbc.integration.Postgres*"`
- `sbt ++2.13 "clean; testOnly org.apache.pekko.persistence.jdbc.integration.Postgres*"`: doesn't run anything 
- `sbt ++2.13 "clean; migratorIntegration/testOnly org.apache.pekko.persistence.jdbc.integration.Postgres*"`: doesn't run anything 

both are unable to run `PostgresJournalMigratorTest`, both the package name and sbt tasks `integration/testOnly` have been verified to cause issues.

#### verify the solution

-  `sbt ++2.13 "integration/testOnly org.apache.pekko.persistence.jdbc.integration.Postgres*; migratorIntegration/testOnly org.apache.pekko.persistence.jdbc.migrator.integration.*"` works, but looks like very long. ✅
-  `sbt ++2.13 "integration/testOnly; migratorIntegration/testOnly org.apache.pekko.persistence.jdbc.integration.Postgres*"` can not filter other RDBMB.

### CODE REVIEW

File: migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/integration/JournalMigratorTest.scala [hunk 1]
```diff
@@ -1,3 +1,22 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
```
Comment (pjfanning) [inline migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/integration/JournalMigratorTest.scala:9]: there is already a license header - we can't add another

File: migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/integration/SnapshotMigratorTest.scala [hunk 2]
```diff
@@ -1,3 +1,22 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
```
Comment (pjfanning) [inline migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/integration/SnapshotMigratorTest.scala:10]: there is already a license header - we can't add another

**Comment by pjfanning** [comment]
As far as I can see, the main issue is that the yml files for the GitHub workflows run `integation/testOnly ...`

`migratorIntegration` is not included in the `root` project so I'm not even sure you can run the tests locally.

File: build.sbt [hunk 3]
```diff
@@ -25,7 +25,7 @@ lazy val `pekko-persistence-jdbc` = project
   .in(file("."))
   .enablePlugins(ScalaUnidocPlugin)
   .disablePlugins(MimaPlugin, SitePlugin)
-  .aggregate(core, migrator, integration, docs)
```
Comment (Roiocam) [inline build.sbt:28]: Is this indeed?

**Comment by Roiocam** [comment]
Seems like `JournalMigratorTest` is incorrect in SQLServer and Oracle:

in SQLServer:

```scala
JournalMigrator(SlickDatabase.profile(config, "slick")).migrate().futureValue shouldBe Done
```

will complain:

```
com.microsoft.sqlserver.jdbc.SQLServerException, with message: Cannot insert explicit value for identity column in table 'event_journal' when IDENTITY_INSERT is set to OFF...
```

File: build.sbt (hunk 3 above)
Comment (pjfanning) [reply build.sbt:28]: looks like it isn't needed

File: migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/migrator/JournalMigratorTest.scala [hunk 4]
```diff
@@ -1,3 +1,22 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
```
Comment (pjfanning) [inline migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/migrator/JournalMigratorTest.scala:7]: this license header is wrong - we already have the license header that says the code is Apache licensed and derived from Akka

File: migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/migrator/MigratorSpec.scala [hunk 5]
```diff
@@ -1,3 +1,22 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
```
Comment (pjfanning) [inline migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/migrator/MigratorSpec.scala:10]: this license header is wrong - we already have the license header that says the code is Apache licensed and derived from Akka

File: migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/migrator/MigratorSpec.scala [hunk 6]
```diff
@@ -9,7 +28,10 @@
 
 package org.apache.pekko.persistence.jdbc.migrator
 
+import com.typesafe.config.{ Config, ConfigFactory, ConfigValue, ConfigValueFactory }
 import org.apache.pekko
```
Comment (pjfanning) [inline migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/migrator/MigratorSpec.scala:32]: could you keep the pekko imports together?

File: migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/migrator/SnapshotMigratorTest.scala [hunk 7]
```diff
@@ -1,3 +1,22 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
```
Comment (pjfanning) [inline migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/migrator/SnapshotMigratorTest.scala:7]: this license header is wrong - we already have the license header that says the code is Apache licensed and derived from Akka

File: migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/migrator/JournalMigratorTest.scala (hunk 4 above)
Comment (Roiocam) [reply migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/migrator/JournalMigratorTest.scala:7]: Sorry, my IDEA automatically did this. I will fix this after investigating Oracle&SQLServer failure.

File: build.sbt (hunk 3 above)
Comment (Roiocam) [reply build.sbt:28]: Thanks, let we keep this change.

**Comment by pjfanning** [comment]
@Roiocam I want to release v1.1.0-M1 of this module within the next week or 2. Would it be possible just to enable the tests for the dbs where this currently works and for us to come back to the broken ones after the v1.1.0-M1 release?

File: .github/workflows/sqlserver-tests.yml [hunk 8]
```diff
@@ -49,7 +49,8 @@ jobs:
         run: sbt ++2.13 "integration/testOnly org.apache.pekko.persistence.jdbc.integration.SqlServer*"  ${{ matrix.extraOpts }} -J-XX:+UnlockExperimentalVMOptions -J-XX:+UseJVMCICompiler
 
       - name: Run Migrator Integration tests for  ${{ matrix.name }}
-        run: sbt ++2.13 "migratorIntegration/testOnly org.apache.pekko.persistence.jdbc.migrator.integration.SqlServer*"  -J-XX:+UnlockExperimentalVMOptions -J-XX:+UseJVMCICompiler
+        if: matrix.name == 'Postgres'
```
Comment (laglangyue) [inline .github/workflows/sqlserver-tests.yml:52]: shoulde be SqlServer?

Comment (Roiocam) [reply .github/workflows/sqlserver-tests.yml:52]: fixed

File: migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/migrator/integration/JournalMigratorTest.scala [hunk 9]
```diff
@@ -19,4 +19,5 @@ class MySQLJournalMigratorTest extends JournalMigratorTest("mysql-application.co
 
 class OracleJournalMigratorTest extends JournalMigratorTest("oracle-application.conf") with OracleCleaner
 
-class SqlServerJournalMigratorTest extends JournalMigratorTest("sqlserver-application.conf") with SqlServerCleaner
+// ignore this because SQLServer didn't support Slick forceInsert
+// class SqlServerJournalMigratorTest extends JournalMigratorTest("sqlserver-application.conf") with SqlServerCleaner
```
Comment (Roiocam) [inline migrator-integration-test/src/test/scala/org/apache/pekko/persistence/jdbc/migrator/integration/JournalMigratorTest.scala:23]: Would you take a look at this? The exception is:

```
Caused by: com.microsoft.sqlserver.jdbc.SQLServerException: 

Cannot insert explicit value for identity column in table 'event_journal' when IDENTITY_INSERT is set to OFF.
```

**Comment by Roiocam** [comment]
@pjfanning This PR is ready for review.

**Review by pjfanning — APPROVED** [review]
lgtm

**Comment by Roiocam** [comment]
@pjfanning Thanks for the review, I also fix SQLServer tests, can you take a look again?

**Comment by pjfanning** [comment]
> @pjfanning Thanks for the review, I also fix SQLServer tests, can you take a look again?

thanks @Roiocam that sqlserver fix looks good.

**Pull request merged**